contracts/governance/Governor.sol

```diff
@@ -29,22 +28,22 @@ import "./IGovernor.sol";
 abstract contract Governor is Context, ERC165, EIP712, IGovernor, IERC721Receiver, IERC1155Receiver {
     using DoubleEndedQueue for DoubleEndedQueue.Bytes32Deque;
     using SafeCast for uint256;
-    using Timers for Timers.BlockNumber;
 
     bytes32 public constant BALLOT_TYPEHASH = keccak256("Ballot(uint256 proposalId,uint8 support)");
     bytes32 public constant EXTENDED_BALLOT_TYPEHASH =
         keccak256("ExtendedBallot(uint256 proposalId,uint8 support,string reason,bytes params)");
 
     struct ProposalCore {
-        Timers.BlockNumber voteStart;
-        Timers.BlockNumber voteEnd;
+        uint256 voteStart; // retyped from Timers.BlockNumber
+        uint256 voteEnd; // retyped from Timers.BlockNumber
```

Comment:
Even though this works currently, I'm not sure it will remain working in future versions of Upgrades Plugins because it doesn't meet our "safety" criteria. We should do something like this:

```suggestion
        uint224 __gap_unused0; // backwards compatibility with <4.9
        uint32 voteStart; // retyped from Timers.BlockNumber
        uint224 __gap_unused1; // backwards compatibility with <4.9
        uint32 voteEnd; // retyped from Timers.BlockNumber
```

Reply:
I think it should be the other way around

```
        uint64 voteStart; // retyped from Timers.BlockNumber
        uint192 __gap_unused0; // backwards compatibility with <4.9
        uint64 voteEnd; // retyped from Timers.BlockNumber
        uint192 __gap_unused1; // backwards compatibility with <4.9
```

Reply:
If I do

```solidity
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.17;

struct Struct1 {
    uint256 value;
}

struct Struct2 {
    uint8 valueBefore;
    uint240 __gap;
    uint8 valueAfter;
}

function getStruct1(bytes32 slot) pure returns (Struct1 storage r) { assembly { r.slot := slot } }
function getStruct2(bytes32 slot) pure returns (Struct2 storage r) { assembly { r.slot := slot } }

contract T {
    bytes32 SLOT = keccak256("some.storage.slot");

    constructor(uint256 value) {
        getStruct1(SLOT).value = value;
    }
    
    function get1() external view returns (Struct1 memory) {
        return getStruct1(SLOT);
    }  

    function get2() external view returns (Struct2 memory) {
        return getStruct2(SLOT);
    }
}
```

And call the constructor with `0xEEFF`

- `get1()` returns `61183`
- `get2()` returns `255,238,0`

Reply:
Similarly

```solidity
struct M1 {
    mapping(uint256 => uint256) value;
}

struct M2 {
    mapping(uint256 => uint64) value;
}

contract T2 {
    bytes32 constant SLOT = keccak256("some.storage.slot");
    uint256 constant KEY = uint256(keccak256("some.mapping.key"));

    function set(uint256 v) public {
        bytes32 slot = SLOT;
        M1 storage ptr;
        assembly { ptr.slot := slot }
        ptr.value[KEY] = v;
    }

    function get() public view returns (uint64) {
        bytes32 slot = SLOT;
        M2 storage ptr;
        assembly { ptr.slot := slot }
        return ptr.value[KEY];
    }
}
```

works